src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -126,6 +130,370 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
+        {
+            string filename = zfile(zipname);
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(filename);
+            string tamperedFileName = "binary.wmv";
+
+            int nameOffset = PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 8);  // patch uncompressed size in file header
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 22, nameOffset + tamperedFileName.Length); // patch in central directory too
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry(tamperedFileName);
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    using (Stream source = e.Open())
+                    {
+                        byte[] buffer = new byte[bufferSize];
+                        int read;
+                        while ((read = source.Read(buffer, 0, buffer.Length)) != 0)
+                        {
+                            ms.Write(buffer, 0, read);
+                        }
+                        Assert.True(e.Length == ms.Length);     // Only allow to decompress up to uncompressed size
+                        Assert.Equal(0, source.Read(buffer, 0, bufferSize)); // shouldn't be able read more 
+                        ms.Seek(0, SeekOrigin.Begin);
+                        while ((read = ms.Read(buffer, 0, buffer.Length)) != 0)
+                        { // No need to do anything, just making sure all bytes readable from output stream
+                        }
+                        Assert.Equal(ms.Position, ms.Length); // all bytes must be read
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Deflate64 zip support is a netcore feature not available on full framework.")]
+        public static async Task Deflate64ZipCorruptedFile_CopyTo()
+        {
+            string filename = compat("deflate64.zip");
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(filename);
+            string tamperedFileName = "binary.wmv";
+
+            int nameOffset = PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 8);  // patch uncompressed size in file header
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 22, nameOffset + tamperedFileName.Length); // patch in central directory too
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry(tamperedFileName);
+                using (var ms = new MemoryStream())
+                {
+                    using (Stream source = e.Open())
+                    {
+                        source.CopyTo(ms);
```

Comment:
Is there a test that, instead of using Stream.CopyTo, does something like

```C#
byte[] buf = new byte[e.Length + 20];
Span<byte> bufSpan = buf.AsSpan();
bufSpan().Fill(0xDE);
byte[] origTrailer = bufSpan.Slice(e.Length).ToArray();
int lastRead = 0;

while ((lastRead = source.Read(bufSpan)) != 0)
{
    bufSpan = bufSpan.Slice(lastRead);
}

Assert.True(buf.AsSpan(e.Length).SequenceEquals(origTrailer), "No extra bytes were written");
```

Reply:
Will add similar check